src/Backends/Firmware.vala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ */
+
+public class SettingsDaemon.Backends.Firmware : GLib.Object {
+    private GLib.Settings firmware_settings;
+
+    private GLib.DateTime last_firmware_check = null;
+
+    private const int SECONDS_BETWEEN_REFRESHES = 60 * 60 * 24;
+
+    public uint firmware_updates_number { get; private set; default = 0U; }
+
+    private AsyncMutex update_notification_mutex = new AsyncMutex ();
+
+    public GLib.Application application { get; construct; }
+
+    public Firmware (GLib.Application application) {
+        Object (
+            application: application
+        );
+    }
+
+    construct {
+        firmware_settings = new GLib.Settings ("io.elementary.settings-daemon.firmware");
+
+        last_firmware_check = new DateTime.from_unix_utc (firmware_settings.get_int64 ("last-refresh-time"));
+
+        check_seconds_between_refreshes ();
+
+        Timeout.add_seconds (SECONDS_BETWEEN_REFRESHES, check_seconds_between_refreshes);
```

Comment:
Should we be using SystemD timers for this?

Reply:
Do you have an example project where we use systemd timer?

Reply:
You basically need an helper executable to be run at intervals.
The ArchLinux documentation is quite good on it: https://wiki.archlinux.org/title/Systemd/Timers

Reply:
In the last commit I created a helper executable, a service and a timer. What is unclear to me is how I can trigger the action in the notification so that the firmware page can be opened in the settings, since the helper executable only runs for a short time. Is it possible to trigger an action from another application via a notification?